lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -326,11 +340,24 @@ sub engine_workit {
         my $error = locate_local_assets(\%vars, $assetkeys);
         return $error if $error;
     }
+    my $casedir     = OpenQA::Utils::testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $productdir  = OpenQA::Utils::productdir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $target_name = path($casedir)->basename;
 
     $vars{ASSETDIR}   //= OpenQA::Utils::assetdir();
-    $vars{CASEDIR}    //= OpenQA::Utils::testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
-    $vars{PRODUCTDIR} //= OpenQA::Utils::productdir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
-
+    $vars{CASEDIR}    //= $target_name;
+    $vars{PRODUCTDIR} //= substr($productdir, rindex($casedir, $target_name));
+
+    my $link_casedir_error = _link_testrepo($casedir, $pooldir, $target_name);
+    return $link_casedir_error if $link_casedir_error;
+
+    if (   $vars{NEEDLES_DIR}
+        && !File::Spec->file_name_is_absolute($vars{NEEDLES_DIR})
+        && $vars{NEEDLES_DIR} !~ /http(s)?\:\/\//)
+    {
+        my $link_needledir_error = _link_testrepo("$productdir/needles", $pooldir, $vars{NEEDLES_DIR});
+        return $link_needledir_error if $link_needledir_error;
+    }
```

Comment:
I just think we do not need to do the symlink because we know os-autoinst will clone the needles and reset the NEEDLES_DIR. And the `symlink("$productdir/needles",  "$pool_dir/https://github.com/xxxx")` will failed, so maybe we skip doing symlink is better.

Reply:
Similar as above: I think you could simply call

```perl
symlink($vars{NEEDLES_DIR}, $pooldir->child('needles')) if $vars{NEEDLES_DIR};
```

and ignore the error cases when a symlink can not be created

Reply:
So as the above comments `failing early and hard is a good approach to find out what is actually to be expected`, could we keep the error handling and `http` check here? I prefer to keep it.  WDYT?  @Martchus

Reply:
I would also avoid ignoring any error blindly, e.g. a permission error might occur and we shouldn't pretend everything is good in this case.

Reply:
ok, fine. But still I think you should be able to simply do a one-line call to symlink pluse one-line error handling and not need a separate function for that. Something like

```perl
my $err = symlink($vars{NEEDLES_DIR}, $pooldir->child('needles')) if $vars{NEEDLES_DIR} && !-e $pooldir->child('needles');
die 'Could not symlink ' . $vars{NEEDLES_DIR} ' into pool directory' if $err;
```

Reply:
I make it a separate function because I could write more details (such as print the debug log message when symlink success) and do more checking (this checking seems to be defensive).  does that way break our coding style? If it does, I do not mind changing my code.  If it does not, I don't think it's necessary to reduce these lines.